compiler/optimizer/GeneralLoopUnroller.cpp

```diff
@@ -1683,13 +1683,63 @@ int32_t TR_LoopUnroller::unroll(TR_RegionStructure *loop, TR_StructureSubGraphNo
     prepareForArrayShadowRenaming(loop);
     refineArrayAliasing();
 
+    bool removedAsyncCheck = false;
+
+    // Remove any asynch check from the original blocks
```

Comment:
I don't think this is quite right. 

We want to remove async checks in the case of a complete unroll. We just don't want to add it back in.

Reply:
Ah I see, I missed the point earlier that there would be no back edge after a loop is completely unrolled, therefore there is no need to add async check in the complete unrolled case I think. I'll update the code and run sanity test tonight. I'll do another push tomorrow once sanity test passes.